File: peek_plugin_search/plugin-module/_private/tuples/SearchIndexChunkTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {searchTuplePrefix} from "../PluginNames";


@addTupleType
export class SearchIndexChunkTuple extends Tuple {
    public static readonly tupleName = searchTuplePrefix + "SearchIndexChunkTuple";

    //  The name of the detail
    name: string;

    //  The value of the detail
    value: string;

    constructor() {
        super(SearchIndexChunkTuple.tupleName)
    }
}

File: peek_plugin_search/_private/tuples/search_index/SearchIndexChunkTuple.py
import logging
from typing import Dict

from peek_plugin_search._private.PluginNames import searchTuplePrefix
from vortex.Tuple import Tuple, addTupleType, TupleField

logger = logging.getLogger(__name__)


@addTupleType
class SearchIndexChunkTuple(Tuple):
    __tupleType__ = searchTuplePrefix + 'SearchIndexChunkTuple'

    #:  This is dict of keywords and objectIds str
    # the dict value format is:
    # {
    #    searchPropertyName: [ objectId, objectId, objectId, ... ]
    # }
    # where searchPropertyName = name, type, *
    objectIdsJsonBykeyword: Dict[str, str] = TupleField()
